src/routes/safe/components/Apps/components/SafeAppCard/SafeAppCard.tsx

```diff
@@ -0,0 +1,238 @@
+import { SyntheticEvent } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { Link } from 'react-router-dom'
+import styled from 'styled-components'
+import { motion } from 'framer-motion'
+import IconButton from '@material-ui/core/IconButton'
+import { alpha } from '@material-ui/core/styles/colorManipulator'
+import { Card, Title, Text, Icon } from '@gnosis.pm/safe-react-components'
+
+import { extractSafeAddress, generateSafeRoute, getShareSafeAppUrl, SAFE_ROUTES } from 'src/routes/routes'
+import { SafeApp } from 'src/routes/safe/components/Apps/types'
+import fallbackSafeAppLogoSvg from 'src/assets/icons/apps.svg'
+import { currentChainId } from 'src/logic/config/store/selectors'
+import enqueueSnackbar from 'src/logic/notifications/store/actions/enqueueSnackbar'
+import { enhanceSnackbarForAction, NOTIFICATIONS } from 'src/logic/notifications'
+import { FETCH_STATUS } from 'src/utils/requests'
+import { copyToClipboard } from 'src/utils/clipboard'
+import { getShortName } from 'src/config'
+import { SafeAppDescriptionSK, SafeAppLogoSK, SafeAppTitleSK } from './SafeAppSkeleton'
+
+type SafeAppCardSize = 'md' | 'lg'
+
+type SafeAppCardProps = {
+  safeApp: SafeApp
+  size: SafeAppCardSize
+  togglePin: (app: SafeApp) => void
+  isPinned?: boolean
+  isCustomSafeApp?: boolean
+  onRemove?: (app: SafeApp) => void
+}
+
+const SafeAppCard = ({
+  safeApp,
+  size,
+  togglePin,
+  isPinned,
+  isCustomSafeApp,
+  onRemove,
+}: SafeAppCardProps): React.ReactElement => {
+  const chainId = useSelector(currentChainId)
+  const dispatch = useDispatch()
+
+  const safeAddress = extractSafeAddress()
+  const appsPath = generateSafeRoute(SAFE_ROUTES.APPS, {
+    shortName: getShortName(),
+    safeAddress,
+  })
+  const openSafeAppLink = `${appsPath}?appUrl=${encodeURI(safeApp.url)}`
+
+  const shareSafeApp = () => {
+    const shareSafeAppUrl = getShareSafeAppUrl(safeApp.url, chainId)
+    copyToClipboard(shareSafeAppUrl)
+    dispatch(enqueueSnackbar(enhanceSnackbarForAction(NOTIFICATIONS.SHARE_SAFE_APP_URL_COPIED)))
+  }
+
+  const isSafeAppLoading = safeApp.fetchStatus === FETCH_STATUS.LOADING
+
+  if (isSafeAppLoading) {
+    return (
+      <SafeAppContainer size={size} layout initial={{ opacity: 0 }} animate={{ opacity: 1 }} exit={{ opacity: 0 }}>
+        <StyledAppCard size={size}>
+          <LogoContainer size={size}>
+            <SafeAppLogoSK size={size} />
+          </LogoContainer>
+          <DescriptionContainer size={size}>
+            <SafeAppTitleSK />
+            <SafeAppDescriptionSK />
+            <SafeAppDescriptionSK />
+          </DescriptionContainer>
+        </StyledAppCard>
+      </SafeAppContainer>
+    )
+  }
+
+  return (
+    <SafeAppContainer size={size} layout initial={{ opacity: 0 }} animate={{ opacity: 1 }} exit={{ opacity: 0 }}>
```

Comment:
Apart from `size`, you're using the same values on both `SafeAppContainer` components. Are we using this container elsewhere or can we abstract them?

Reply:
mmm what do you mean with abstract them?

Reply:
Perhaps just have them fixed in the component so we don't have to pass them via props at all, or create a `safeContainProps` and spread them onto both. What do you think?

Reply:
do you mean this?
```
const SafeAppContainer = styled(motion.div).attrs({
  layout: true,
  initial: { opacity: 0 },
  animate: { opacity: 1 },
  exit: { opacity: 0 },
})`
  position: relative;
  display: flex;
```

Reply:
Yes, exactly.

Reply:
done!